### DISCUSSION TITLE
Problem changing Network channel with zigpy 56.0

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#1224] [Q&A]
I'm not able to change the Network channel since migrating to zigpy 56.0 and bellows 0.35.5

I guess the API has changed from
```
        self.config[bellows_conf.CONF_NWK][bellows_conf.CONF_NWK_CHANNEL] = channel
        await self._ezsp.leaveNetwork()
        await super().form_network()
```

> 2023-06-16 15:40:15,559 INFO    :b"RAW-COMMAND: {'Function' : zdp_raw_nwk_update_request,'Profile' : 0000,'Cluster' : 0038,'TargetNwk' : 0000,'TargetEp' : 00,'SrcEp' : 00,'Sqn' : 00,'RxOnIdle' : False,'payload' : 0000080000fe01,'timestamp' : 1686922814.9999452,'AddressMode' : 02,}"
> 2023-06-16 15:40:15,565 INFO    :b'ZigyTransport: process_raw_command ready to request Function: zdp_raw_nwk_update_request NwkId: 0000/0 Cluster: 0038 Seq: 59 Payload: 0000080000fe01 AddrMode: 02 EnableAck: True, Sqn: 0, Delay: None, Extended_TO: False'
> 2023-06-16 15:40:15,567 DEBUG   :Sending: b'75e421a9602a15b259944a1daa5592099c4e27f2b7c9678bf5c66377fd7b6e7e'
> 2023-06-16 15:40:15,568 INFO    :b"process_raw_command  call request destination: <Device model='EZSP' manuf='Silicon Labs' nwk=0x0000 ieee=84:b4:db:ff:fe:f8:2f:cd is_initialized=True> Profile: 0 Cluster: 56 sEp: 0 dEp: 0 Seq: 89 Payload: b'\\x00\\x00\\x08\\x00\\x00\\xfe\\x01'"
> 2023-06-16 15:40:15,572 INFO    :b'got a command REQ-NWK-STATUS'
> 2023-06-16 15:40:15,573 INFO    :b"transport_request: _limit_concurrency <Device model='EZSP' manuf='Silicon Labs' nwk=0x0000 ieee=84:b4:db:ff:fe:f8:2f:cd is_initialized=True> 89"
> 2023-06-16 15:40:15,574 INFO    :b"ZigyTransport: process_raw_command  84:b4:db:ff:fe:f8:2f:cd 0 (<class 'int'>) 56 (<class 'int'>) - Max Attemps: 1"
> 2023-06-16 15:40:15,579 DEBUG   :Data frame: b'50e4a1a9602a1565ddbf7e'
> 2023-06-16 15:40:15,580 DEBUG   :Sending: b'8610be7e'
> 2023-06-16 15:40:15,583 DEBUG   :Application frame received sendUnicast: [<EmberStatus.SUCCESS: 0>, 215]
> 2023-06-16 15:40:15,587 DEBUG   :Data frame: b'60e4b1a9112a15b259ac4a25aa1592499c99d8abedce9874fac66381fc7ec1a687047e'
> 2023-06-16 15:40:15,588 DEBUG   :Sending: b'87009f7e'
> 2023-06-16 15:40:15,592 DEBUG   :Application frame received incomingMessageHandler: [<EmberIncomingMessageType.INCOMING_UNICAST: 0>, EmberApsFrame(profileId=0, clusterId=56, sourceEndpoint=0, destinationEndpoint=0, options=<EmberApsOption.APS_OPTION_RETRY: 64>, groupId=0, sequence=215), 255, 0, 0x0000, 255, 255, b'\x00\x00\x08\x00\x00\xfe\x01']
> 2023-06-16 15:40:15,595 DEBUG   :Data frame: b'70e4b1a96b2a15b259944a1daa5592099c4e277cb7ce676c5a7e'
> 2023-06-16 15:40:15,595 DEBUG   :Received incomingMessageHandler frame with [<EmberIncomingMessageType.INCOMING_UNICAST: 0>, EmberApsFrame(profileId=0, clusterId=56, sourceEndpoint=0, destinationEndpoint=0, options=<EmberApsOption.APS_OPTION_RETRY: 64>, groupId=0, sequence=215), 255, 0, 0x0000, 255, 255, b'\x00\x00\x08\x00\x00\xfe\x01']
> 2023-06-16 15:40:15,596 DEBUG   :Sending: b'8070787e'
> 2023-06-16 15:40:15,598 DEBUG   :Received a packet: ZigbeePacket(src=AddrModeAddress(addr_mode=<AddrMode.NWK: 2>, address=0x0000), src_ep=0, dst=AddrModeAddress(addr_mode=<AddrMode.NWK: 2>, address=0x0000), dst_ep=0, source_route=None, extended_timeout=False, tsn=215, profile_id=0, cluster_id=56, data=Serialized[b'\x00\x00\x08\x00\x00\xfe\x01'], tx_options=<TransmitOptions.NONE: 0>, radius=0, non_member_radius=0, lqi=255, rssi=0)
> 2023-06-16 15:40:15,606 DEBUG   :[0x0000:zdo] ZDO request ZDOCmd.Mgmt_NWK_Update_req: [NwkUpdate(ScanChannels=<Channels.CHANNEL_11: 2048>, ScanDuration=254, nwkUpdateId=1)]
> 2023-06-16 15:40:15,608 INFO    :b'handle_message from Controller Sender: 0x0000 Profile: 0000 Cluster: 0038 srcEp: 00 dstEp: 00 message: 0000080000fe01'
> 2023-06-16 15:40:15,610 DEBUG   :[0x0000:zdo] No handler for ZDO request:ZDOCmd.Mgmt_NWK_Update_req([NwkUpdate(ScanChannels=<Channels.CHANNEL_11: 2048>, ScanDuration=254, nwkUpdateId=1)])
> 2023-06-16 15:40:15,614 INFO    :b'handle_message Sender: 0000 frame for plugin: 0180020028ff000000003800000200000200000000080000fe01ff03'
> 2023-06-16 15:40:15,620 DEBUG   :Application frame received messageSentHandler: [<EmberOutgoingMessageType.OUTGOING_DIRECT: 0>, 0, EmberApsFrame(profileId=0, clusterId=56, sourceEndpoint=0, destinationEndpoint=0, options=<EmberApsOption.APS_OPTION_RETRY: 64>, groupId=0, sequence=215), 90, <EmberStatus.SUCCESS: 0>, b'']
> 2023-06-16 15:40:15,625 DEBUG   :Received messageSentHandler frame with [<EmberOutgoingMessageType.OUTGOING_DIRECT: 0>, 0, EmberApsFrame(profileId=0, clusterId=56, sourceEndpoint=0, destinationEndpoint=0, options=<EmberApsOption.APS_OPTION_RETRY: 64>, groupId=0, sequence=215), 90, <EmberStatus.SUCCESS: 0>, b'']
> 2023-06-16 15:40:15,629 INFO    :b'ZigyTransport: process_raw_command completed 89 NwkId: 0000 result: Status.SUCCESS'
> 2023-06-16 15:40:16,057 DEBUG   :Data frame: b'00e4b1a96b2a134fa6944a1daa5592499c4e277db5a867fabf7e'
> 2023-06-16 15:40:16,058 DEBUG   :Sending: b'8160597e'
> 2023-06-16 15:40:16,063 DEBUG   :Application frame received messageSentHandler: [<EmberOutgoingMessageType.OUTGOING_BROADCAST: 6>, 65533, EmberApsFrame(profileId=0, clusterId=56, sourceEndpoint=0, destinationEndpoint=0, options=<EmberApsOption.APS_OPTION_NONE: 0>, groupId=0, sequence=214), 88, <EmberStatus.DELIVERY_FAILED: 102>, b'']
> 2023-06-16 15:40:16,065 DEBUG   :Received messageSentHandler frame with [<EmberOutgoingMessageType.OUTGOING_BROADCAST: 6>, 65533, EmberApsFrame(profileId=0, clusterId=56, sourceEndpoint=0, destinationEndpoint=0, options=<EmberApsOption.APS_OPTION_NONE: 0>, groupId=0, sequence=214), 88, <EmberStatus.DELIVERY_FAILED: 102>, b'']
> 2023-06-16 15:40:16,067 DEBUG   :Unexpected message send notification tag: 88

**Comment by pipiche38** [comment]
I guess we need to use that 

https://github.com/zigpy/zigpy/blob/a3e449582fe2a1203316dce0955ccfe19d6fdc3b/zigpy/application.py#LL261C5-L262C31
```
    async def move_network_to_channel(
        self, new_channel: int, *, num_broadcasts: int = 5
    ) -> None:
```

**Reply by pipiche38** [reply]
Looks like it is not enought as the network do not switch

> 2023-06-16 16:14:10,856 DEBUG   :[0x0000:zdo] ZDO request ZDOCmd.Mgmt_NWK_Update_req: [NwkUpdate(ScanChannels=<Channels.CHANNEL_15: 32768>, ScanDuration=254, nwkUpdateId=1)]
> 2023-06-16 16:14:10,857 INFO    :b'handle_message from Controller Sender: 0x0000 Profile: 0000 Cluster: 0038 srcEp: 00 dstEp: 00 message: 0000800000fe01'
> 2023-06-16 16:14:10,860 DEBUG   :[0x0000:zdo] No handler for ZDO request:ZDOCmd.Mgmt_NWK_Update_req([NwkUpdate(ScanChannels=<Channels.CHANNEL_15: 32768>, ScanDuration=254, nwkUpdateId=1)])
> 2023-06-16 16:14:10,865 INFO    :b'handle_message Sender: 0000 frame for plugin: 0180020028ff000000003800000200000200000000800000fe01ff03'
> 2023-06-16 16:14:10,871 DEBUG   :Application frame received messageSentHandler: [<EmberOutgoingMessageType.OUTGOING_DIRECT: 0>, 0, EmberApsFrame(profileId=0, clusterId=56, sourceEndpoint=0, destinationEndpoint=0, options=<EmberApsOption.APS_OPTION_RETRY: 64>, groupId=0, sequence=5), 56, <EmberStatus.SUCCESS: 0>, b'']
> 2023-06-16 16:14:10,876 DEBUG   :Received messageSentHandler frame with [<EmberOutgoingMessageType.OUTGOING_DIRECT: 0>, 0, EmberApsFrame(profileId=0, clusterId=56, sourceEndpoint=0, destinationEndpoint=0, options=<EmberApsOption.APS_OPTION_RETRY: 64>, groupId=0, sequence=5), 56, <EmberStatus.SUCCESS: 0>, b'']
> 2023-06-16 16:14:10,879 INFO    :b'ZigyTransport: process_raw_command completed 55 NwkId: 0000 result: Status.SUCCESS'
> 2023-06-16 16:14:11,203 DEBUG   :Data frame: b'04b8b1a96b2a134fa6944a1daa5592499c4e27afdbce678bc17e'
> 2023-06-16 16:14:11,204 DEBUG   :Sending: b'8160597e'
> 2023-06-16 16:14:11,208 DEBUG   :Application frame received messageSentHandler: [<EmberOutgoingMessageType.OUTGOING_BROADCAST: 6>, 65533, EmberApsFrame(profileId=0, clusterId=56, sourceEndpoint=0, destinationEndpoint=0, options=<EmberApsOption.APS_OPTION_NONE: 0>, groupId=0, sequence=4), 54, <EmberStatus.SUCCESS: 0>, b'']
> 2023-06-16 16:14:11,209 DEBUG   :Received messageSentHandler frame with [<EmberOutgoingMessageType.OUTGOING_BROADCAST: 6>, 65533, EmberApsFrame(profileId=0, clusterId=56, sourceEndpoint=0, destinationEndpoint=0, options=<EmberApsOption.APS_OPTION_NONE: 0>, groupId=0, sequence=4), 54, <EmberStatus.SUCCESS: 0>, b'']
> 2023-06-16 16:14:11,211 DEBUG   :Unexpected message send notification tag: 54
> 2023-06-16 16:14:15,252 DEBUG   :Send command readCounters: ()
> 2023-06-16 16:14:15,261 DEBUG   :Sending: b'41b921a9a52aa2b67e'
> 2023-06-16 16:14:15,280 DEBUG   :Data frame: b'15b9a1a9a52a4db21b947125835596499c4e22abe8ce4c8be2c66789fe7e3ca7e9cdde6f8fffc7dbd5d2698c4623a9ec763ba5ea758241984c2617b1e070381c0e07bbe5ca658a459a4d9e4f9ff7c3d9d46a35a251904824d1097e'
> 2023-06-16 16:14:15,281 DEBUG   :Sending: b'82503a7e'
> 2023-06-16 16:14:15,286 DEBUG   :Application frame received readCounters: [[88, 66, 59, 41, 4, 0, 5, 5, 43, 31, 4, 2, 3, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]]
> 2023-06-16 16:14:15,289 DEBUG   :Send command getValue: (<EzspValueId.VALUE_FREE_BUFFERS: 3>,)
> 2023-06-16 16:14:15,292 DEBUG   :Sending: b'52be21a9fe2a16d8d87e'
> 2023-06-16 16:14:15,299 DEBUG   :Data frame: b'26bea1a9fe2a15b3a0edd27e'
> 2023-06-16 16:14:15,300 DEBUG   :Sending: b'83401b7e'
> 2023-06-16 16:14:15,303 DEBUG   :Application frame received getValue: [<EzspStatus.SUCCESS: 0>, b'\xf9']
> 2023-06-16 16:14:15,305 DEBUG   :Free buffers status EzspStatus.SUCCESS, value: 249
> 2023-06-16 16:14:15,306 DEBUG   :ezsp_counters: [MAC_RX_BROADCAST = 88, MAC_TX_BROADCAST = 66, MAC_RX_UNICAST = 59, MAC_TX_UNICAST_SUCCESS = 41, MAC_TX_UNICAST_RETRY = 4, MAC_TX_UNICAST_FAILED = 0, APS_DATA_RX_BROADCAST = 5, APS_DATA_TX_BROADCAST = 5, APS_DATA_RX_UNICAST = 43, APS_DATA_TX_UNICAST_SUCCESS = 31, APS_DATA_TX_UNICAST_RETRY = 4, APS_DATA_TX_UNICAST_FAILED = 2, ROUTE_DISCOVERY_INITIATED = 3, NEIGHBOR_ADDED = 2, NEIGHBOR_REMOVED = 0, NEIGHBOR_STALE = 0, JOIN_INDICATION = 0, CHILD_REMOVED = 0, ASH_OVERFLOW_ERROR = 0, ASH_FRAMING_ERROR = 0, ASH_OVERRUN_ERROR = 0, NWK_FRAME_COUNTER_FAILURE = 0, APS_FRAME_COUNTER_FAILURE = 0, UTILITY = 0, APS_LINK_KEY_NOT_AUTHORIZED = 0, NWK_DECRYPTION_FAILURE = 0, APS_DECRYPTION_FAILURE = 4, ALLOCATE_PACKET_BUFFER_FAILURE = 0, RELAYED_UNICAST = 0, PHY_TO_MAC_QUEUE_LIMIT_REACHED = 0, PACKET_VALIDATE_LIBRARY_DROPPED_COUNT = 0, TYPE_NWK_RETRY_OVERFLOW = 0, PHY_CCA_FAIL_COUNT = 0, BROADCAST_TABLE_FULL = 0, PTA_LO_PRI_REQUESTED = 0, PTA_HI_PRI_REQUESTED = 0, PTA_LO_PRI_DENIED = 0, PTA_HI_PRI_DENIED = 0, PTA_LO_PRI_TX_ABORTED = 0, PTA_HI_PRI_TX_ABORTED = 0, ADDRESS_CONFLICT_SENT = 0, EZSP_FREE_BUFFERS = 249]
> 2023-06-16 16:14:20,818 INFO    :b'Zigbee Coordinator ieee: 84b4dbfffef82fcd , short addr: 0000'
> 2023-06-16 16:14:20,819 INFO    :b'Network state UP, PANID: b254 extPANID: 0x790e890f9a03f4c6 Channel: 11'

**Reply by puddly** [reply]
You seem to be calling `_move_network_to_channel`, not `move_network_to_channel`. Your debug log doesn't contain any zigpy logging from the latter.
